ddclient.in

```diff
@@ -6401,42 +6401,50 @@ sub nic_hetzner_update {
                     next;
                 }
                 # Pull the ID out of the json, messy
-                my ($dns_rec_id) = map { ($_->{name} eq $hostname && $_->{type} eq $type) ? $_->{id} : ()} @{$response->{records}};
+                my ($dns_rec_id, $current_ip) = map { ($_->{name} eq $hostname && $_->{type} eq $type) ? ($_->{id}, $_->{value}) : ()} @{$response->{records}};
 
-                # Set domain
                 my $http_method="";
-                if ($dns_rec_id)
-                {
-                    debug("updating %s: DNS '$type' record ID: $dns_rec_id", $domain);
-                    $url   = "https://$config{$key}{'server'}/records/$dns_rec_id";
-                    $http_method = "PUT";
-                } else {
-                    debug("creating %s: DNS '$type'", $domain);
-                    $url   = "https://$config{$key}{'server'}/records";
-                    $http_method = "POST";
-                }
-                my $data = "{\"zone_id\":\"$zone_id\", \"name\": \"$hostname\", \"value\": \"$ip\", \"type\": \"$type\", \"ttl\": $config{$domain}{'ttl'}}";
-
-                $reply = geturl(proxy => opt('proxy'),
-                                url => $url,
-                                headers => $headers,
-                                method => $http_method,
-                                data => $data
-                               );
-                unless ($reply && header_ok($domain, $reply)) {
-                    failed("updating %s: Could not connect to %s.", $domain, $config{$domain}{'server'});
-                    next;
-                }
-                # Strip header
-                $reply =~ qr/{(?:[^{}]*|(?R))*}/mp;
-                $response = eval {decode_json(${^MATCH})};
-                if ($response && $response->{record}) {
-                    success("updating %s: IPv$ipv address set to %s", $domain, $ip);
+                if ($current_ip && $current_ip eq $ip) {
+                    info("updating %s %s: IP is already %s, no update needed.", $hostname, $ipv, $ip);
```

Comment:
well there isn't a documented rate limit..., but i wanted to take the load from the api

Reply:
- Some providers require periodic updates even if the IP address has not changed, otherwise they will delete the DNS record. I'm not familiar with Hetzner; do they not require occasional refreshes?
  - Hetzner does not have dedicated DynDNS-Entries, you edit Records directly over their [API](https://dns.hetzner.com/api-docs)
- Infrequent refreshes are useful for periodically testing that ddclient and the credentials still work. Does Hetzner prohibit this?
  - Dont think so, but for reading a record, the api token is also validated

- Is ddclient frequently setting the same IP over and over? It should not do that. If it is, then either the config is inappropriate (e.g., max-interval is set too short) or there's a cache bug in the hetzner logic that should be fixed.